core/src/scene/styleParam.cpp

```diff
@@ -152,62 +120,70 @@ StyleParamKey StyleParam::getKey(const std::string& _key) {
     return it->second;
 }
 
-StyleParam::StyleParam(const std::string& _key, const std::string& _value) {
+StyleParam::StyleParam(const std::string& _key, const YAML::Node& _value) {
     key = getKey(_key);
     value = none_type{};
 
     if (key == StyleParamKey::none) {
-        LOGW("Unknown StyleParam %s:%s", _key.c_str(), _value.c_str());
+        LOGW("Unknown StyleParam %s:%s", _key.c_str(), Dump(_value).c_str());
         return;
     }
-    if (!_value.empty()) {
-        value = parseString(key, _value);
-    }
+    value = parseNode(key, _value);
+}
+
+StyleParam::StyleParam(StyleParamKey _key, const YAML::Node& _value) {
+    key = _key;
+    value = parseNode(key, _value);
 }
 
-StyleParam::Value StyleParam::parseString(StyleParamKey key, const std::string& _value) {
-    auto allowedUnits = unitsForStyleParam(key);
+
+StyleParam::Value StyleParam::parseNode(StyleParamKey key, const YAML::Node& node) {
+    UnitSet allowedUnits = unitSetForStyleParam(key);
 
     switch (key) {
     case StyleParamKey::extrude: {
-        return parseExtrudeString(_value);
+        return parseExtrudeNode(node);
     }
     case StyleParamKey::text_wrap: {
-        int textWrap;
-        if (_value == "false") {
-            return std::numeric_limits<uint32_t>::max();
-        }
-        if (_value == "true") {
-            return uint32_t(15); // DEFAULT
-        }
-        if (parseInt(_value, textWrap) > 0 && textWrap > 0) {
-             return static_cast<uint32_t>(textWrap);
+        bool useWrap = false;
+        int wrapLength = 15; // Default wrap length.
+        if (YAML::convert<bool>::decode(node, useWrap) && useWrap) {
+            return (uint32_t)wrapLength;
+        } else if (YAML::convert<int>::decode(node, wrapLength) && wrapLength > 0) {
+            return (uint32_t)wrapLength;
         }
         return std::numeric_limits<uint32_t>::max();
     }
     case StyleParamKey::text_offset:
     case StyleParamKey::offset: {
+        ValueUnitPair scalar;
+        if (node.IsScalar() && (parseValueUnitPair(node.Scalar(), scalar) && allowedUnits.contains(scalar.unit))) {
+            return glm::vec2(scalar.value);
+        }
         UnitVec<glm::vec2> vec;
-        if (!parseVec2(_value, allowedUnits, vec) || std::isnan(vec.value.y)) {
-            LOGW("Invalid offset parameter '%s'.", _value.c_str());
+        if (parseVec2(node, allowedUnits, vec)) {
+            return vec.value;
         }
-        return vec.value;
+        LOGW("Invalid offset parameter '%s'.", Dump(node).c_str());
+        return Value();
     }
     case StyleParamKey::text_buffer:
     case StyleParamKey::buffer: {
-        UnitVec<glm::vec2> vec;
-        if (!parseVec2(_value, allowedUnits, vec)) {
-            LOGW("Invalid buffer parameter '%s'.", _value.c_str());
+        ValueUnitPair scalar;
```

Comment:
This block of close for `text_buffer`/`buffer` is same as `text_offset`/`offset`. Maybe just club these together.

Reply:
Combined these blocks, seems to work just fine.